File: src/test/java/br/com/controle/financeiro/controller/linkbuilder/TransactionDTOResourceAssemblerTest.java
package br.com.controle.financeiro.controller.linkbuilder;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.hateoas.Resource;
import org.springframework.test.context.junit4.SpringRunner;

import br.com.controle.financeiro.model.dto.TransactionDTO;

@RunWith(SpringRunner.class)
@SpringBootTest(classes = { TransactionDTOResourceAssembler.class })
@AutoConfigureMockMvc
public class TransactionDTOResourceAssemblerTest {

	@InjectMocks
	private TransactionDTOResourceAssembler resourceAssembler;

	@Test
	public void testToResource() {
		TransactionDTO transactionMock = new TransactionDTO();
		transactionMock.setId(1L);
		Resource<TransactionDTO> response = resourceAssembler.toResource(transactionMock);
		assertTrue(response.hasLinks());
		assertNotNull(response.getLink("self"));
		assertNotNull(response.getLink("transactions"));
		assertEquals(transactionMock, response.getContent());
	}

}

File: src/test/kotlin/br/com/controle/financeiro/controller/api/linkbuilder/TransactionDTOResourceAssemblerTest.kt
package br.com.controle.financeiro.controller.api.linkbuilder

import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles
import org.springframework.beans.factory.annotation.Autowired
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions
import br.com.controle.financeiro.model.dto.TransactionDTO
import br.com.controle.financeiro.controller.api.tra

@SpringBootTest
@ActiveProfiles(profiles = ["test"])
class TransactionDTOResourceAssemblerTest(
    @Autowired private val resourceAssembler: TransactionDTOResourceAssembler
) {
    @Test
    fun testToResource() {
        val transactionMock = TransactionDTO(tra.name, tra.transactionDate, tra.value, tra.currency, tra.responsible.id, tra.bankAccount.id, tra.card.id, tra.id)
        val response = resourceAssembler.toModel(transactionMock)
        Assertions.assertTrue(response.hasLinks())
        Assertions.assertNotNull(response.getLink("self"))
        Assertions.assertNotNull(response.getLink("transactions"))
        Assertions.assertEquals(transactionMock, response.content)
    }
}
